arango/connection.py

```diff
@@ -110,6 +110,42 @@ def prep_response(self, resp: Response, deserialize: bool = True) -> Response:
         resp.is_success = http_ok and resp.error_code is None
         return resp
 
+    def process_response(
+        self, host_index: int, request: Request, auth: Optional[Tuple[str, str]] = None
+    ) -> Response:
+        """Execute a request until a valid response has been returned.
+
+        :param host_index: The index of the first host to try
+        :type host_index: int
+        :param request: HTTP request.
+        :type request: arango.request.Request
+        :return: HTTP response.
+        :rtype: arango.response.Response
+        """
+        tries = 0
+        while tries < self._host_resolver._max_tries:
+            try:
+                resp = self._http.send_request(
+                    session=self._sessions[host_index],
+                    method=request.method,
+                    url=self._url_prefixes[host_index] + request.endpoint,
+                    params=request.params,
+                    data=self.normalize_data(request.data),
+                    headers=request.headers,
+                    auth=auth,
+                )
+
+                return self.prep_response(resp, request.deserialize)
+            except ConnectionError:
```

Comment:
We are not preserving the contents of ConnectionErrors (e.g. error messages) here, which may hinder debugging. A few options I can think of are 1) calling `logger.exception` 2) pass down the contents of ConnectionErrors in ConnectionAbortedError (seems like the more difficult option).

Reply:
Right, I'm not 100% sure what you're looking for, but I've added [this line](https://github.com/ArangoDB-Community/python-arango/pull/184/files#diff-ff65f7eb9f247417f86b688f08c1dc65226bfe8298866e685349f3a902c062bdR142) in our `except` clause. Can you confirm that this is what you were thinking of having?

Reply:
Yes that's correct.

Reply:
I'll switch this to `logging.error` for now to avoid the noise generated by `logging.exception`

Reply:
Good point. But did you mean to say `logging.debug`? Usually we use lower level messages to allow users to reduce noise.

Reply:
Ah good call, I've updated that